azurerm/internal/services/postgres/postgresql_server_resource.go

```diff
@@ -64,13 +64,30 @@ func resourcePostgreSQLServer() *schema.Resource {
 
 		Importer: &schema.ResourceImporter{
 			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
-				if _, err := parse.ServerID(d.Id()); err != nil {
+				client := meta.(*clients.Client).Postgres.ServersClient
+				ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+				defer cancel()
+
+				id, err := parse.ServerID(d.Id())
+				if err != nil {
 					return []*schema.ResourceData{d}, err
 				}
 
+				resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+				if err != nil {
+					return []*schema.ResourceData{d}, fmt.Errorf("reading PostgreSQL Server %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+				}
+
 				d.Set("create_mode", "Default")
-				if v, ok := d.GetOk("create_mode"); ok && v.(string) != "" {
-					d.Set("create_mode", v)
+				d.Set("creation_source_server_id", "")
+				if *resp.ReplicationRole != "Master" && *resp.ReplicationRole != "None" {
```

Comment:
we'll need to nil check ReplicationRole to prevent panics

Reply:
Thanks, done!